### ISSUE TITLE
Add exception for all json files in workflows and add schema validation

### ISSUE THREAD

**Issue by olefasting** [fishfight/FishFight#187]
We should add an exception for changes to json files for all our checks on PRs and add schema validation for all files.
Currently, this only applies for maps and we will, in the case of maps, have to figure out a way to differentiate between tiled maps and non-tiled maps. This is already done in the game, with a variable in the map declarations file but if we need to add some other criteria for workflows, that is fine. We could, for example, add an inner extension to tiled maps (filename.tiled.json), if it is not possible to add parameters for deducing this from the data in the data file. If multiple schemas can be used on the same file, I can add a parameter to the internal map format to accommodate for this...

I am also aiming at finishing my work on items and all the dynamic loading of resources, either this weekend or early next week, and this will introduce a whole lot of new data files that should be validated. Once this PR is merged, I will help whoever takes this on making schema files for all of this...

**Comment by orhun** [comment]
> Is there any way to have dependencies across schemas, though?

Not sure if we can do that.

> So, maybe the best thing would be to do the PoC on the decalaration files (`textures.json`, `sounds.json`, etc)

Sure, it sounds like a good start.

I actually had some progress about this. I decided use the following tools:

- https://extendsclass.com/json-schema-validator.html (for generating [JSON Schema Draft-07](https://json-schema.org/draft-07/json-schema-release-notes.html), we can use any other similar tool for this purpose)
- [ajv-cli](https://github.com/ajv-validator/ajv-cli/) (for schema validation)

I'm thinking of having separate schema files in the same directory as the JSON files, like this:

```
assets/
├── sounds.json
└── sounds.schema.json
```

Then in a simple bash script, we can give `find` these schemas and validate them with `ajv`.

For the directories, we can have only one schema and we can validate the files inside the directory using it:

```
assets/
└── items
    ├── cannon.json
    ├── grenades.json
    ├── kick_bomb.json
    ├── machine_gun.json
    ├── mines.json
    ├── musket.json
    ├── sniper_rifle.json
    ├── sword.json
    ├── turtle_shell.json
    └── schema.json
```

The validation can be handled in one bash script and we can just run that in a separate workflow I think.

Of course there might be some drawbacks and edges cases but this is the cleanest solution that I could have thought of so far 🙂

**Comment by olefasting** [comment]
Sorry, for the late reply, your answer slipped by me. This all sounds good to me!
